README.md

```diff
@@ -5,24 +5,61 @@ clr-boot-manager
 [![Coverage Status](https://coveralls.io/repos/github/ikeydoherty/clr-boot-manager/badge.png?branch=master)](https://coveralls.io/github/ikeydoherty/clr-boot-manager?branch=master)
 
 
-clr-boot-manager  exists  to  enable the correct maintainence of vendor
-kernels and appropriate garbage collection tactics over the  course  of
-upgrades.   The  implementation  provides  the  means to enable correct
-cohabitation on a shared boot directory, such as the EFI System  Partition
-for UEFI-booting operating systems.
+clr-boot-manager  exists  to  enable the correct maintenance of vendor kernels and appropriate garbage collection tactics over the  course  of upgrades.   The  implementation  provides  the  means to enable correct cohabitation on a shared boot directory, such as the EFI System  Partition for UEFI-booting operating systems.
 
-Special  care  is taken to ensure the ESP is handled gracefully, and in
-the instance that it is not already mounted, then clr-boot-manager will
-automatically  discover and mount it, and automatically unmount the ESP
-again when it is complete.
+Special  care  is taken to ensure the ESP is handled gracefully, and in the instance that it is not already mounted, then clr-boot-manager will automatically  discover and mount it, and automatically unmount the ESP again when it is complete.
+
+Most importantly, clr-boot-manager provides a simple mechanism to provide kernel updates, with the ability for users to rollback to an older kernel should the new update be problematic. This is achieved through the use of strict namespace policies, permanent source paths, and clr-boot-manager's own internal logic, without the need for "meta packages" or undue complexity on the distribution side.
 
 Requirements
 ------------
 
-clr-boot-manager is designed to operate solely with `GPT` disks, and
-exclusively uses `PARTUUID`. Generated boot entries also contain the `PARTUUID`
-in their `root=` command line, as part of a merge of the vendor provided
-`cmdline` files for default options.
+clr-boot-manager is primarily designed for GPT disks using UEFI, however it does contain fallback support for legacy bootloaders such as GRUB2 to allow all users to benefit from automated kernel management.
+
+clr-boot-manager should be the only tool responsible within the OS for generating boot entries, and will automatically incorporate the correct `root=` portions.
+
+Kernel Integration
+------------------
+
+The way that a kernel is packaged changes significantly with clr-boot-manager. First and foremost, no files shall be shipped in `/boot`. The distribution should choose a namespace to identify their system in dual-boot situations, i.e:
+
+    org.someproject
+
+All paths known to CBM must have follow a specific format and encoding, whereby
+the version, release number, and *type* are encoded:
+
+    /usr/lib/kernel
+     -> config-4.9.17-9.lts
```

Comment:
Hrm so we should probably have all objects in a shared dir be namespaced it seems at some point.